src/core/ext/filters/client_channel/channel_connectivity.cc

```diff
@@ -161,15 +161,15 @@ class StateWatcher {
       }
       switch (phase_) {
         case kWaiting:
-          GRPC_ERROR_REF(error);
+          (void)GRPC_ERROR_REF(error);
```

Comment:
This should just be combined with the next line:

```
error_ = GRPC_ERROR_REF(error);
```

Note that I've just rewritten this code in #27747, so once that gets merged, this will go away anyway.